### ISSUE TITLE
Catch exception type of Exception and process body request

### ISSUE THREAD

**Issue by Hedgehogues** [tiangolo/fastapi#575]
**Description**

How can I catch exception type of Exception and process body request?

**Additional context**

My example

@app.exception_handler(Exception)
async def exception_callback(request: requests.Request, exc: Exception):
    request_json = await request.json() # Here i want a body, query_params, headers and path of request
    print(request_json)

@app.post('/create_workflow')
async def create_workflow():
    raise Exception(13)

**Comment by euri10** [comment]
mmm I'm not sure, but I don't think you can consume body in a middleware (see https://github.com/tiangolo/fastapi/issues/394)

getting the other parameters you mentioned is possible though (note that the `Request` below is from `starlette` and not `requests` I think you imported the wrong one), but should you need the body that won't happen

I'd ask in Starlette if I were you

```
import pytest
from async_asgi_testclient import TestClient
from fastapi import FastAPI
from starlette.requests import Request

app = FastAPI()


@app.exception_handler(Exception)
async def exception_callback(request: Request, exc: Exception):
    print(request.query_params)
    print(request.headers)
    print(request.path_params)
    print(exc.args)


@app.post("/create_workflow")
async def create_workflow():
    raise Exception(13)


@pytest.mark.asyncio
async def test_raise_exc():
    async with TestClient(app) as client:
        with pytest.raises(Exception) as pe:
            response = await client.post("/create_workflow")
```

**Comment by Hedgehogues** [comment]
Yes, I saw it. But this code cannot extract body of request. Example of printing:


    ```
print(request.query_params)
    print(request.headers)
    print(request.path_params)
    print(exc.args)
```

**Comment by euri10** [comment]
yes as I pointed out in the above linked issue #394, this is not something you can achieve in a middleware.
you might want to read https://github.com/encode/starlette/issues/495#issuecomment-494008175 as well and ask on Starlette

**Comment by tiangolo** [comment]
A new way to handle these cases was added in #589.

The docs are here: https://fastapi.tiangolo.com/tutorial/custom-request-and-route/
